### PR TITLE
Clarify align_to's requirements and obligations

### PR DESCRIPTION
**Pull request by shepmaster**
Based on discussions with @RalfJung around my misunderstandings when using this.

r? @RalfJung

### CODE REVIEW

**Comment by RalfJung** [comment]
Thanks!

@bors r+ rollup

File: src/libcore/slice/mod.rs [hunk 1]
```diff
@@ -2308,9 +2308,10 @@ impl<T> [T] {
     /// maintained.
     ///
     /// This method splits the slice into three distinct slices: prefix, correctly aligned middle
-    /// slice of a new type, and the suffix slice. The method does a best effort to make the
-    /// middle slice the greatest length possible for a given type and input slice, but only
-    /// your algorithm's performance should depend on that, not its correctness.
+    /// slice of a new type, and the suffix slice. The method may make the middle slice the greatest
+    /// length possible for a given type and input slice, but only your algorithm's performance
+    /// should depend on that, not its correctness. It is permissable for all of the input data to
```
Comment (Centril) [inline src/libcore/slice/mod.rs]:
```suggestion
    /// should depend on that, not its correctness. It is permissible for all of the input data to
```

File: src/libcore/slice/mod.rs [hunk 2]
```diff
@@ -2361,9 +2362,10 @@ impl<T> [T] {
...
```
Comment (Centril) [inline src/libcore/slice/mod.rs]:
```suggestion
    /// should depend on that, not its correctness. It is permissible for all of the input data to
```

**Comment by Centril** [comment]
@bors r-

Some typos :)

**Comment by shepmaster** [comment]
Thanks!

@bors r=@RalfJung

**Comment by RalfJung** [comment]
No `@` for bors (yeah don't ask)

@bors r+

**Pull request merged**